test/distributed/_sharded_optim/test_sharded_optim.py

```diff
@@ -89,6 +99,13 @@ def test_sharded_optim(self):
             else:
                 self.assertNotEqual(val, new_val)
 
+        # run second forward with updated parameters
```

Comment:
nit: Any reason to do a second forward instead of just comparing the local and sharded parameters are the same?

Reply:
no particular reason haha, just thought it was easier to do that, updated to compare the weights directly